std-bits/table/src/main/java/org/enso/table/excel/ExcelRow.java

```diff
@@ -37,7 +37,12 @@ public Object getCellValue(int column) {
         if (DateUtil.isCellDateFormatted(cell)) {
           return cell.getLocalDateTimeCellValue().toLocalDate();
         } else {
-          return cell.getNumericCellValue();
+          double dblValue = cell.getNumericCellValue();
+          if (dblValue == (long) dblValue) {
+            return (long) dblValue;
+          } else {
+            return dblValue;
+          }
```

Comment:
No Excel has no difference - or at least POI doesn't.

Reply:
Ok. That's a bummer.

I thought maybe we could use the decimal places from the formatting? i.e. I think format allows to set how many decimal places are displayed and it can also be 0 for integers. BUT this only changes the way the data is displayed, not the underlying data IIRC, so I'm not sure if this would be correct, likely not quite.